test/functional-e2e/identity-flows/mail-page-object.ts

```diff
@@ -0,0 +1,38 @@
+import { clickVisibleElement } from '@test/utils/ui.test.helper';
+import puppeteer from 'puppeteer';
+
+const lastEmail = '#mailItemsColumn tr:first-child a[href="#"]';
+const redirectLink = '#mailDetailsColumn p:nth-child(4) a';
+const mailRefreshButton = '#btnRefresh';
+const mailRows = '#mailItemsColumn tr';
+const loadingIndicator = '.blockUI.blockMsg';
+
+export default class MailPage {
+  page: puppeteer.Page | undefined;
+  value: string | undefined;
+
+  static async clickRefreshButton(page: puppeteer.Page) {
+    await clickVisibleElement(page, mailRefreshButton);
+    await page.waitForSelector(loadingIndicator, {
+      visible: false,
+      hidden: true,
+    });
+  }
+
+  static async openLastEmail(page: puppeteer.Page) {
+    await clickVisibleElement(page, lastEmail);
+    await page.waitForSelector(loadingIndicator, {
```

Comment:
`waitForLoadingIndicatorToHide` ?

Reply:
Will make a common function to be used for both Alkemio and Mailslurper.